[AutoJMBridge] Rework of the script to handle the new website

sysadminstory opened the pull request:
The brand - model page does not have filters now, so the actual feed
using this bridge are broken.

Website has changed. The new website offers a new search function with
many more filter. So I switched to this new search page for this bridge.

Fixes #2372

sysadminstory committed:
[AutoJMBridge] Rework of the script to handle the new website

The brand - model page does not have filters now, so the actual feed
using this bridge are broken.

Website has changed. The new website offers a new search function with
many more filter. So I switched to this new search page for this bridge.

em92 commented:
For some reason phpunit7 stopped running. Previous attempt could not run phpunit for 44(!) minutes https://github.com/RSS-Bridge/rss-bridge/runs/4633946789?check_suite_focus=true

em92 commented on bridges/AutoJMBridge.php:
```diff
@@ -75,71 +36,102 @@ public function getName() {
 		// Get the JSON content of the form
-		$json = getContents($model_url . '?' . $get_data, $header)
+		$json = getContents($search_url, $header)
 			or returnServerError('Could not request AutoJM.');
```
please remove returnServerError here. It is useless, since getContents raises exception on error

sysadminstory replied:
As requested, removed in 92a90ff :)

em92 commented on bridges/AutoJMBridge.php:
```diff
@@ -75,71 +36,102 @@ public function getName() {
+
+		// Get the HTML content of the page
+		$html = getSimpleHTMLDOMCached($search_url)
+			or returnServerError('Could not request AutoJM.');
```
Also remove here returnServerError

sysadminstory replied:
As requested, removed in 92a90ff :)

em92 reviewed (changes requested):
Hi, @sysadminstory!
Please fix code below. Don't mention phpunit7 errors.

sysadminstory committed:
[AutoJMBridge] Rework of the script to handle the new website

Remove unnecessary returnServerError() after the getContents() and
getSimpleHTMLDOMCached() function
